python/hsfs/engine/python.py

```diff
@@ -797,18 +797,47 @@ def add_file(self, file):
         # can be used to materialize certificates locally
         return file
 
-    @staticmethod
-    def _apply_transformation_function(transformation_functions, dataset):
+    def _apply_transformation_function(self, transformation_functions, dataset):
         for (
             feature_name,
             transformation_fn,
         ) in transformation_functions.items():
             dataset[feature_name] = dataset[feature_name].map(
                 transformation_fn.transformation_fn
             )
+            dataset[feature_name] = self.infer_python_type(
+                transformation_fn.output_type, dataset[feature_name]
+            )
 
         return dataset
 
+    @staticmethod
+    def infer_python_type(output_type, feature_column):
```

Comment:
Is there a particular reason for using `.apply()`? `.astype()` is the better choice here performance wise. Look at the following short example

```
import pandas as 
import time
import numpy as np
data = {
        "primary_key": range(1,1000000)
    }

df = pd.DataFrame(data)

#  0.1617434024810791 s
df["primary_key"].apply(float)

# 0.0019593238830566406 s
df["primary_key"].astype(float)